vm/src/vm/runners/cairo_runner.rs

```diff
@@ -517,9 +527,9 @@ impl CairoRunner {
         &self.program.builtins
     }
 
-    fn consumed(&self, run_resources: &mut Option<RunResources>) -> bool {
-        if let Some(r) = run_resources.as_ref() {
-            r.consumed()
+    fn consumed(&self, run_resources: &mut RunResources) -> bool {
+        if run_resources.consume_steps {
+            run_resources.consumed()
         } else {
             false
         }
```

Comment:
This method seems redundant given `RunResources::consumed`

Reply:
true